configdb/hardware/serializers.py

```diff
@@ -106,19 +106,35 @@ class Meta:
         model = Camera
 
 
+class InstrumentTypeSerializer(serializers.ModelSerializer):
+    mode_types = GenericModeGroupSerializer(many=True)
+
+    class Meta:
+        fields = ('id', 'name', 'code', 'fixed_overhead_per_exposure',
+                  'front_padding', 'config_change_time', 'acquire_exposure_time',
+                  'mode_types', 'default_acceptability_threshold',
+                  'allow_self_guiding', 'configuration_types')
+        model = InstrumentType
+
+
 class InstrumentSerializer(serializers.ModelSerializer):
     science_camera = CameraSerializer(read_only=True)
     science_camera_id = serializers.IntegerField(write_only=True)
     autoguider_camera = CameraSerializer(read_only=True)
     autoguider_camera_id = serializers.IntegerField(write_only=True)
     telescope = serializers.HyperlinkedRelatedField(view_name='telescope-detail', read_only=True)
     telescope_id = serializers.IntegerField(write_only=True)
+    science_cameras = CameraSerializer(read_only=True, many=True)
+    science_cameras_ids = serializers.PrimaryKeyRelatedField(write_only=True, many=True, queryset=Camera.objects.all(), source='science_cameras')
+    instrument_type = InstrumentTypeSerializer(read_only=True)
+    instrument_type_id = serializers.IntegerField(write_only=True)
```

Comment:
Why are the `instrument_type_id` and `science_cameras_ids` set to write_only?

Reply:
The _id fields are only used for creating a new object by POSTing data. It takes an integer _id for the foreign key relationship rather than a fully defined object for it, because it assumes it already exists. The non _id fields are read_only because they are used for the output from the API, so it shows the full object representation rather than the foreign key id.

Reply:
Oh right, I forgot that we have endpoints to update configuration.